### PR TITLE
-PerformancePlus for New-AzDiskConfig

### PR DESCRIPTION
**Pull request by grizzlytheodore**
## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Compute/Compute/Generated/Disk/Config/NewAzureRmDiskConfigCommand.cs [hunk 1]
```diff
@@ -242,6 +242,12 @@ public partial class NewAzureRmDiskConfigCommand : Microsoft.Azure.Commands.Reso
         [PSArgumentCompleter("X64", "Arm64")]
         public string Architecture { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Set this flag to true to get a boost on the performance target of the disk deployed, see here on the respective performance target. This flag can only be set on disk creation time and cannot be disabled after enabled.")]
+        public bool? PerformancePlus { get; set; }
```
Comment (BethanyZhou) [inline src/Compute/Compute/Generated/Disk/Config/NewAzureRmDiskConfigCommand.cs:249]: We'd like suggest to use switch parameter for PerformancePlus in New-Az* command

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
Suppress false positive for 
```
  D:\a\_work\1\s\artifacts//StaticAnalysisResults\SignatureIssues.csv Errors
  "Module","ClassName","Target","Severity","ProblemId","Description","Remediation"
  "Az.Compute","Microsoft.Azure.Commands.Compute.Automation.NewAzureRmDiskConfigCommand","New-AzDiskConfig","1","8410","Parameter PerformancePlus of cmdlet New-AzDiskConfig does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
  ```

File: src/Compute/Compute/Generated/Disk/Config/NewAzureRmDiskConfigCommand.cs (hunk 1 above)
Comment (grizzlytheodore) [reply src/Compute/Compute/Generated/Disk/Config/NewAzureRmDiskConfigCommand.cs:249]: yes, switchParameter would make more sense but all the boolean properties for Disk in this cmdlet [New-AzDiskConfig](https://learn.microsoft.com/en-us/powershell/module/az.compute/new-azdiskconfig?view=azps-9.4.0) are defined as boolean and not Switch. So I wanted to keep it consistent within this cmdlet. 
I think it's better than having one boolean property be switch while ~3 others are all boolean for this cmdlet.

Comment (grizzlytheodore) [reply src/Compute/Compute/Generated/Disk/Config/NewAzureRmDiskConfigCommand.cs:249]: @BethanyZhou

**Comment by grizzlytheodore** [comment]
> Suppress false positive for
> 
> ```
>   D:\a\_work\1\s\artifacts//StaticAnalysisResults\SignatureIssues.csv Errors
>   "Module","ClassName","Target","Severity","ProblemId","Description","Remediation"
>   "Az.Compute","Microsoft.Azure.Commands.Compute.Automation.NewAzureRmDiskConfigCommand","New-AzDiskConfig","1","8410","Parameter PerformancePlus of cmdlet New-AzDiskConfig does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
> ```

thank you. suppressed

**Review by BethanyZhou — APPROVED** [review]
looks good to me

**Pull request merged**